api-template-java/Jenkinsfile

```diff
@@ -1,18 +1,46 @@
 #!groovy
 import groovy.json.JsonOutput
-node {
+import groovy.transform.Field
+
+// To be replaced as @Field def repo_credential_id = "value" except for config_loader
+@Field def repo_credential_id
+@Field def repo_base
+@Field def repo_core
+@Field def config_loader
 
-withCredentials([[$class: 'UsernamePasswordMultiBinding', credentialsId: env.JENKINS_CREDENTIAL_ID, passwordVariable: 'PWD', 
-usernameVariable: 'UNAME']]){
-    
-   echo "Build triggered via branch::${env.BRANCH_NAME}"
-   echo "params : $params"
-  
-   def build_job = env.API_BUILD_URI
-   def var_job_url = JenkinsLocationConfiguration.get().getUrl() + build_job
-  
-    sh "curl -X GET -k -v -u \"$UNAME:$PWD\"  \"" + var_job_url + "&service_name={service_name}&domain={domain}&scm_branch=${env.BRANCH_NAME}\""
+/*
+* Load environment variables from build module
+*/
+def load_config_module(build_module_url){
+  echo "loading environment variables, checking repos..."
+
+  dir('config-loader') {
+    checkout([$class: 'GitSCM', branches: [[name: '*/master']], doGenerateSubmoduleConfigurations: false, extensions: [], submoduleCfg: [], userRemoteConfigs: [[credentialsId: repo_credential_id, url: build_module_url]]])
+    echo "loading installer variables..."
+    def result_json_string = readFile("jazz-installer-vars.json")
+    config_module = load "config-loader.groovy"
+    config_loader = config_module.initialize(result_json_string)
+    echo "Finished loading environment module"
+  }
 }
-    
-  
+
+/*
+* Start the Pipeline stages
+*/
+node {
+  stage('Loading Config-Loader') {
+    build_module_url = "http://" + repo_base + "/scm/" + repo_core + "/jazz-build-module.git"
```

Comment:
use convention of single string with variables => "http://${repo_base}/scm/${repo_core}/jazz-build-module.git"

Reply:
completed.